### PR TITLE
feat: `dbAppendTable()` accepts `Id`

### LINKED ISSUE #253: `dbAppendTable(con, Id(...), data)` no longer works
`Id(schema = "foo", table = "bar")` could be used as `name` argument in a number of functions, as documented at https://github.com/r-dbi/DBI/blob/main/R/Id.R. However, the latest release no longer seems to support `Id` at all. Whenever I supply a `Id(...)`, the following error occurs, say that it must be a character vector.

```
Error in dbAppendTable(con, id, data) : 
  is.character(name) is not TRUE
```

Am I missing something?

### PR DESCRIPTION
**Pull request by renkun-ken**
Closes #253 

This will work if the upstream https://github.com/r-dbi/DBI/pull/381 is also merged.

### CODE REVIEW

File: tests/testthat/test-dbWriteTable.R [hunk 1]
```diff
@@ -32,6 +32,14 @@ test_that("dbAppendTable() throws error if constraint violated", {
   expect_error(dbAppendTable(con, "t1", x), "added 7 rows|Duplicate entry")
 })
 
+test_that("dbAppendTable() works with Id", {
+  con <- mariadbDefault()
+  on.exit(dbDisconnect(con))
+
+  dbExecute(con, "create table T(n integer primary key)")
```
Comment (krlmlr) [inline tests/testthat/test-dbWriteTable.R:39]: Do we want a temporary table here?

Comment (renkun-ken) [reply tests/testthat/test-dbWriteTable.R:39]: Just notice that the table will persist. Should I drop it in the end of the test case, or drop table if exists before creating it?

Comment (renkun-ken) [reply tests/testthat/test-dbWriteTable.R:40]: I think drop table if exists makes good sense as it is similar with `dbWriteTable(overrite=TRUE)` in other test cases.

Comment (krlmlr) [reply tests/testthat/test-dbWriteTable.R:40]: `CREATE TEMPORARY TABLE` might be easier. Ideally all test cases should leave a clean state in the database.

Comment (renkun-ken) [reply tests/testthat/test-dbWriteTable.R]: Yes, you are right.

**Comment by renkun-ken** [comment]
The test fails because it has to rely on https://github.com/r-dbi/DBI/pull/381.

**Comment by renkun-ken** [comment]
@krlmlr Any suggestions on this?

**Comment by krlmlr** [comment]
Thanks!

**Pull request merged**